### PR TITLE
Fix icalexport with new versions of plone.app.event.

### PR DESCRIPTION
**Pull request by pcdummy**
Signed-off-by: Rene Jochum rene@jochums.at

### CODE REVIEW

File: Solgema/fullcalendar/browser/icalexport.py [hunk 1]
```diff
@@ -63,7 +64,7 @@ def update(self):
                 query = {'object_provides': IEvent.__identifier__}
             else:
                 query = {'portal_type': 'Event'}
-            self.events = context.queryCatalog(**query)
+            self.events = api.content.find(context=context, **query)
```
Comment (frisi) [inline Solgema/fullcalendar/browser/icalexport.py]: collection.queryCatalog takes the collection criteria into account and adds additional query parameters. 
this way the collection criteria get lost and you limit the results to the path of the collection

Comment (pcdummy) [inline Solgema/fullcalendar/browser/icalexport.py]: Ty for that info!

With context.queryCatalog i get an unknown parameter 'object_provides' or 'portal_type', do you know howto solve that @frisi ?

File: Solgema/fullcalendar/browser/icalexport.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def update(self):
                 query = {'object_provides': IEvent.__identifier__}
             else:
                 query = {'portal_type': 'Event'}
-            self.events = context.queryCatalog(**query)
+            self.events = context.queryCatalog(query)
```
Comment (pcdummy) [inline Solgema/fullcalendar/browser/icalexport.py]: Now back using context.queryCatalog as @frisi pointed out.

**Comment by hvelarde** [comment]
guys, it's important to add a build in CI to test this in Plone 4.3 with and without plone.app.event installed; check https://github.com/collective/collective.fingerpointing/blob/master/src/collective/fingerpointing/testing.py for some ideas on how to implement this.

IMO, support for Plone < 4.3 can be removed.

**Comment by frisi** [comment]
thanks for your feedback @hvelarde 
i'm afraid, fixing the test-infrastructure for this package is out of our scope for this project.
we just needed to fix the ical export feature and hope for a merge and maybe even a small release @thet ?

**Comment by hvelarde** [comment]
@frisi I know tests are somehow broken, but you must be sure your changes are not affecting anybody else: your tests must pass on both configurations.

**Comment by thet** [comment]
LGTM
+ WOW! the rename of construct_calendar to construct_icalendar was done pre plone.app.event 1.0:
https://github.com/plone/plone.app.event/commit/ef574bfaf9a672a6545a5b7db64d3e9d8273a45b

Sorry for the late reply, but I absolutly missed this PR.

**Pull request merged**